Tests/UnitTests/Core/Vertexing/VertexingDataHelper.cpp

```diff
@@ -0,0 +1,157 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <fstream>
+#include <iterator>
+#include <regex>
+
+#include "Acts/Tests/CommonHelpers/DataDirectory.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Units.hpp"
+#include "Acts/Vertexing/Vertex.hpp"
+
+namespace Acts {
+namespace Test {
+
+using namespace Acts::UnitLiterals;
+using Covariance = BoundSymMatrix;
+
+// Create a test context
+GeometryContext geoCtx = GeometryContext();
+
+enum VertexCsvData { BeamSpotData, VerticesData, TracksData };
+
+/// @brief Helper struct to store reference vertex related information
+struct VertexInfo {
+  // The position
+  Vector3D position;
+  // The covariance
+  ActsSymMatrixD<3> covariance;
+  // Number of tracks
+  int nTracks;
+  // Weight of first track
+  double trk1Weight;
+  // Vertex compatibility value of first track
+  double trk1Comp;
+  // Chi2 of first track
+  double trk1Chi2;
+};
+
+std::tuple<Vertex<BoundParameters>, std::vector<VertexInfo>,
+           std::vector<BoundParameters>>
+readTracksAndVertexCSV(std::string toolString,
+                       std::string fileBase = "vertexing_event_mu20") {
+  const auto tracksDataPath = Acts::Test::getDataPath(fileBase + "_tracks.csv");
+  const auto verticesDataPath =
+      Acts::Test::getDataPath(fileBase + "_vertices_" + toolString + ".csv");
+
+  const std::regex comma(",");
+
+  // Open source files
+  std::ifstream tracksData(tracksDataPath);
+  std::ifstream verticesData(verticesDataPath);
+
+  // We want to read all lines of the file
+  std::string line{};
+  bool isBeamSpot = false;
```

Comment:
This is not a robust/ scalable way to handle the different data files. I still think it would be better to split the data into separate files that are strict CSV files, i.e. a header line that defines the number of columns and their names followed by additional data lines that have exactly the same number of entries as the header. This is the format that is typically produces and read by other software, e.g. numpy, and using it would also easily compatibility (e.g. if you want to generate fake data for testing). Since this is used in a test we should avoid having custom formats.

You could also have a single function that reads the data and then just interpret it differently for the different types of data. Since you know what each file should contain, you can easily check e.g. that the files have the right number of columns and you do not need the extra `bool`s to distinguish between the different formats.

Reply:
Changed.